src/event.js

```diff
@@ -120,6 +120,11 @@ jQuery.event = {
 			selector = handleObjIn.selector;
 		}
 
+		// If the selector is invalid, throw any exceptions at attach time
+		if ( selector ) {
+			jQuery.find( selector, elem );
```

Comment:
This may or may not require a context `elem.ownerDocument`. If it's just testing validity, perhaps not, but can anyone think of a use case where the document would affect validity?